spec/lib/workflow/stores/mongo-spec.js

```diff
@@ -619,32 +599,262 @@ describe('Task Graph mongo store interface', function () {
         });
     });
 
+    it('updates dependent tasks under anyOf key', function() {
+        var waterlineUnderAnyOf = {
+            taskdependencies:{
+                find: sinon.stub().resolves([]),
+                //make sure the find cannot find anything under dependent.taskId
+                updateMongo: sinon.stub().resolves()
+            }
+        };
+
+        helper.setupInjector([
+            helper.require('/lib/workflow/stores/mongo'),
+            helper.di.simpleWrapper(waterlineUnderAnyOf, 'Services.Waterline')
+        ]);
+        mongo = helper.injector.get('TaskGraph.Stores.Mongo');
+        waterline = helper.injector.get('Services.Waterline');
+
+        var data = {
+            state: 'succeeded',
+            taskId: uuid.v4(),
+            graphId: uuid.v4()
+        };
+
+        return mongo.updateDependentTasks(data)
+        .then(function() {
+            var query = {
+                graphId: data.graphId,
+                reachable: true
+            };
+
+            query['dependencies.anyOf.' + data.taskId] = {
+                $in: [data.state, Constants.Task.States.Finished]
+            };
+
+            var updateAny = {
+                $unset:{}
+            };
+            updateAny.$unset['dependencies.anyOf'] = "";
+
+            expect(waterline.taskdependencies.updateMongo).to.have.been.calledOnce;
+            expect(waterline.taskdependencies.updateMongo).to.have.been.calledWith(
+                query, updateAny, {multi: true});
+        });
+    });
+
     it('updateUnreachableTasks', function() {
```

Comment:
This description need be modified since there are still other test cases are testing this function with different condition.